### ISSUE TITLE
CYGWIN Failing: Terminus Requires Psysh?

### ISSUE THREAD

**Issue by grok** [pantheon-systems/documentation#3916]
I have cygwin installed:

```
$ php -v
PHP 7.1.16 (cli) (built: Apr 10 2018 18:09:30) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
```
```
$ curl --version
curl 7.59.0 (x86_64-unknown-cygwin) libcurl/7.59.0 OpenSSL/1.0.2o zlib/1.2.11 libidn2/2.0.4 libpsl/0.18.0 (+libidn2/2.0.2) libssh2/1.7.0 nghttp2/1.31.0
Release-Date: 2018-03-14
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS Debug IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL Metalink
```

But when I try to install Terminus:

```
$ terminus
-bash: terminus: command not found
```
```
$ ls
composer.json  installer.phar
````
```
$ php installer.phar install
Installing Terminus...
Using version ^1.8 for pantheon-systems/terminus
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - pantheon-systems/terminus 1.8.0 requires psy/psysh ^0.8 -> satisfiable by psy/psysh[v0.8.0, v0.8.1, v0.8.10, v0.8.11, v0.8.12, v0.8.13, v0.8.14, v0.8.15, v0.8.16, v0.8.1                                                                                                                 7, v0.8.18, v0.8.2, v0.8.3, v0.8.4, v0.8.5, v0.8.6, v0.8.7, v0.8.8, v0.8.9].
    - pantheon-systems/terminus 1.8.1 requires psy/psysh ^0.8 -> satisfiable by psy/psysh[v0.8.0, v0.8.1, v0.8.10, v0.8.11, v0.8.12, v0.8.13, v0.8.14, v0.8.15, v0.8.16, v0.8.1                                                                                                                 7, v0.8.18, v0.8.2, v0.8.3, v0.8.4, v0.8.5, v0.8.6, v0.8.7, v0.8.8, v0.8.9].
    - psy/psysh v0.8.0 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.1 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.10 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.11 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.12 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.13 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.14 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.15 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.16 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.17 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.18 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.                                                                                                                 0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.2 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.3 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.4 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.5 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.6 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.7 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.8 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - psy/psysh v0.8.9 requires nikic/php-parser ~1.3|~2.0|~3.0 -> satisfiable by nikic/php-parser[v1.3.0, v1.4.0, v1.4.1, v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, v3.0.1, v3.0                                                                                                                 .2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.1.0, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5].
    - nikic/php-parser v3.1.5 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.1.4 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.1.3 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.1.2 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.1.1 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.1.0 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.6 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.5 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.4 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.3 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.2 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.1 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v3.0.0 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v2.1.1 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v2.1.0 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v2.0.1 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v2.0.0 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v1.4.1 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v1.4.0 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - nikic/php-parser v1.3.0 requires ext-tokenizer * -> the requested PHP extension tokenizer is missing from your system.
    - Installation request for pantheon-systems/terminus ^1.8 -> satisfiable by pantheon-systems/terminus[1.8.0, 1.8.1].

  To enable extensions, verify that they are enabled in your .ini files:
    -
    - /etc/php.d/ctype.ini
    - /etc/php.d/curl.ini
    - /etc/php.d/gd.ini
    - /etc/php.d/json.ini
    - /etc/php.d/mbstring.ini
    - /etc/php.d/mysqli.ini
    - /etc/php.d/pdo_mysql.ini
    - /etc/php.d/phar.ini
    - /etc/php.d/posix.ini
  You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.

Installation failed, reverting ./composer.json to its original content.
Terminus was installed, but the installer was not able to write to your bin dir. To enable the
`terminus` command, add this alias to your .bash_profile (Mac) or .bashrc (Linux) file:

alias terminus=/home/sterling/vendor/bin/terminus

Or you can enable it by adding the directory the executable file is in to your path:

PATH="/home/sterling/vendor/bin:$PATH"
```

It says it installed... but that folder doesn't exist:
```
$ ls /home/sterling/
composer.json  installer.phar
```

**Comment by grok** [comment]
Also, it's probably separate thing that it says "I FAILED!" and then it goes "EVERYTHING IS FINE!" ... and everything is nooottt fine.

**Comment by alexfornuto** [comment]
CC @sehamilton @greg-1-anderson

**Comment by greg-1-anderson** [comment]
Presently, there are no Windows configurations that work with Terminus without issues. Until these bugs are fixed, the best thing to do is to run Terminus under Linux using a virtual machine.

**Comment by carl-alberto** [comment]
@greg-1-anderson Do you have a recommended guide where we can at least point out to frustrated Windows users in setting up their VM with minimal version of Linux to at least run Terminus with? Would it be a good additional info in the Terminus installation guide too https://pantheon.io/docs/terminus/install/ ?

**Comment by greg-1-anderson** [comment]
I do not know a good pre-existing source for that. It would in fact be useful to have Windows instructions in our documentation.

I did happen to just notice that the problem originally reported was likely called by running the Terminus installer in a directory that already had another application installed in the composer.json / vendor directory there. Always create a new empty directory before running the Terminus installer.

**Comment by alexfornuto** [comment]
OK. Let's wrap up this issue.

@grok: Per @greg-1-anderson, try installing Terminus in a fresh folder without any existing composer files/folders.

@carl-alberto: Ubuntu makes a good distro for someone's first test into linux. 

@greg-1-anderson I recall you've said something about a Docker image for Terminus before. Would that make more sense than a Linux VM for Windows users? I attempted to make a minimal image using Alpine Linux, but have so far been unable to make it work.


Closing this issue as it's not leading to an update to the docs, but I'm happy to continue the discussion.
